tests/privacy_scenario_3_test.py

```diff
@@ -0,0 +1,290 @@
+    Print("API Node1 currently at {}".format(apiNode1_head))
+    Print("producer 0 after security group transaction block became LIB: {}".format(prod1_head))
+    # non-producing nodes do not receive the block that moved lib to or past the recorded block
+    Print("API Node1 head = {} Prod1 head = {}".format(apiNode1_head, prod1_head))
+    assert apiNode1_head <= prod1_head
+
+    trans = securityGroup.editSecurityGroup(addNodes=[apiNode1])
+    # record block number (rbn1) transaction was added in
+    trans_block_num = prod1.getBlockNumByTransId(trans["transaction_id"], blocksAhead=3)
+    assert trans_block_num is not None
+    Print("security group update [[\"node3\"]] published on block {} or after".format(trans_block_num))
+    securityGroup.verifyParticipantsTransactionFinalized(trans["transaction_id"])
+
+    cluster.launchUnstarted(cachePopen=True)
+    # waiting to make sure that the launched node has had a chance to start so we know it isn't syncing
+    prod1.waitForBlock(prod1.getHeadBlockNum() + 25)
+    apiNode2 = cluster.getNode(3)
+    # after last recorded block number (rbn2) is made irreversible, verify
+    # starting node2 results in connection being made, but no blocks or transactions being sent
+    Print("API Node2 head block = {}".format(apiNode2.getHeadBlockNum()))
+    assert apiNode2.getHeadBlockNum() == 1
+    
+    apiNode1ListenEP = apiNode1.getListenEndpoint()
+    apiNode1ListenEP.replace("localhost", "0.0.0.0")
+    Print("API Node1 p2p-listen-endpoint is {}".format(apiNode1ListenEP))
+
+    foundConnection = False
+    for connection in apiNode2.getConnections():
+        p2p_address = connection["last_handshake"]["p2p_address"]
+        match = re.search("([a-z\.0-9]+:[0-9]{2,5}) ", p2p_address)
+        if match and len(match.groups()):
+            cur_address = match.groups()[0].replace("localhost", "0.0.0.0")
+            Print("current p2p_address = " + cur_address)
+            if cur_address == str(apiNode1ListenEP):
+                foundConnection = True
+                break
+    
+    assert foundConnection
+
+    # add node2 to security group *
+    trans = securityGroup.editSecurityGroup(addNodes=[apiNode2])
+    # ecord block number (rbn3) transaction was added in
+    trans_block_num = prod1.getBlockNumByTransId(trans["transaction_id"], blocksAhead=3)
+    assert trans_block_num is not None
+    Print("security group update [[\"node4\"]] published on block {} or above".format(trans_block_num))
+
+    # before last recorded block number (rbn3) is made irreversible, verify
+    # node2 still receives no blocks or transactions
+    breakTimeout = 100
+    while breakTimeout > 0:
+        apiNode2Head = apiNode2.getHeadBlockNum()
+        apiNode1Lib  = apiNode1.getIrreversibleBlockNum()
+        if apiNode1Lib >= trans_block_num:
+            break
+        Print("API Node2 head block = {}".format(apiNode2Head))
+        assert apiNode2Head == 1
+        apiNode1.waitForNextBlock(sleepTime=0.5)
```

Comment:
assert apiNode1.waitforNextBlock(sleepTime=0.5) != None

this will just allow us to fail faster if apiNode1 head isn't moving